engine/api/workflow_run.go

```diff
@@ -1032,6 +1058,121 @@ func (api *API) initWorkflowRun(ctx context.Context, projKey string, wf *sdk.Wor
 	}, api.PanicDump())
 }
 
+func saveWorkflowRunSecrets(ctx context.Context, db gorp.SqlExecutor, projID int64, wr sdk.WorkflowRun, secrets *workflow.PushSecrets) error {
+	// Get project secrets
+	p, err := project.LoadByID(db, projID, project.LoadOptions.WithVariablesWithClearPassword, project.LoadOptions.WithClearKeys)
+	if err != nil {
+		return err
+	}
+
+	// Filter on secret and keys
+	pv := sdk.VariablesFilter(p.Variables, sdk.SecretVariable, sdk.KeyVariable)
+	// Add prefix
+	pv = sdk.VariablesPrefix(pv, "cds.proj.")
+	for _, v := range pv {
+		wrSecret := sdk.WorkflowRunSecret{
+			WorkflowRunID: wr.ID,
+			Context:       "proj",
+			Name:          v.Name,
+			Value:         []byte(v.Value),
+		}
+		if err := workflow.InsertRunSecret(ctx, db, &wrSecret); err != nil {
```

Comment:
It could be nice to insert all the secrets in ons sql request

Reply:
not supported by gorp